### ISSUE TITLE
ENH: condense decorators e.g. @cython.cfunc(returns=foo, nogil=True, exceptval=-1)

### ISSUE THREAD

**Issue by jbrockmendel** [cython/cython#3714]
Looking at converting some cython-syntax code to be valid python, what is one line in cython becomes much more verbose in pure-python mode:

```
cdef bint is_leapyear(int64_t year) nogil:

@cython.nogil
@cython.returns(bint)
@cython.cfunc
def is_leapyear(year: int64_t):
```

The decorators here could reasonably be condensed to something like `@cython.cfunc(nogil=True, returns=bint)`

(ideally the `@cython.returns(bint)` could be specified python-style by `-> bint:`, but thats a separate issue)

**Comment by scoder** [comment]
Yeah, the `@cython.returns()` isn't really needed any more, now that return type annotations can be used.
That reduces the gain to `@cython.cfunc(nogil=True)`. Is that still worth it?

**Comment by jbrockmendel** [comment]
In this example its less of a big deal, but there's also exceptval and potentially wraparound and boundscheck
